components/core/tests/test-ir_serializer.cpp

```diff
@@ -0,0 +1,147 @@
+#include <chrono>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp/ErrorCode.hpp"
+#include "../src/clp/ffi/ir_stream/decoding_methods.hpp"
+#include "../src/clp/FileReader.hpp"
+#include "../src/clp/ir/constants.hpp"
+#include "../src/clp/ir/LogEventDeserializer.hpp"
+#include "../src/clp/ir/LogEventSerializer.hpp"
+#include "../src/clp/streaming_compression/zstd/Decompressor.hpp"
+#include "../src/clp/Utils.hpp"
+
+using clp::ir::eight_byte_encoded_variable_t;
+using clp::ir::four_byte_encoded_variable_t;
+using std::string;
+using std::vector;
+
+using clp::ffi::ir_stream::deserialize_log_event;
+
+using std::chrono::milliseconds;
+using std::chrono::system_clock;
+using std::is_same_v;
+
+using clp::ErrorCode_Success;
+using clp::ffi::ir_stream::IRErrorCode::IRErrorCode_Success;
+using clp::FileReader;
+using clp::ir::cIrFileExtension;
+using clp::ir::epoch_time_ms_t;
+using clp::ir::LogEventDeserializer;
+using clp::ir::LogEventSerializer;
+using clp::streaming_compression::zstd::Decompressor;
+
+namespace {
+template <typename encoded_variable_t>
+auto match_encoding_type(bool is_four_bytes_encoding) -> bool {
+    static_assert(
+            (is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>)
+            || (is_same_v<encoded_variable_t, four_byte_encoded_variable_t>)
+    );
+
+    if constexpr (is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>) {
+        return false == is_four_bytes_encoding;
+    } else {
+        return is_four_bytes_encoding;
+    }
+}
+}  // namespace
+
+/*
+ * The test case only covers four byte encoding because decompressor
+ * does not support eight bytes encoding yet.
+ */
```

Comment:
1. I don't understand what do u mean by `decompressor does not support eight bytes encoding`. It works locally if I enable `eight_byte_encoding_t`
2. We should probably move the comment inside the template body
3. 100 char per line